include/zephyr/net/wifi.h

```diff
@@ -245,4 +245,131 @@ static inline const char *wifi_link_mode_txt(enum wifi_link_mode link_mode)
 	}
 }
 
+enum wifi_ps {
+	WIFI_PS_DISABLED = 0,
+	WIFI_PS_ENABLED,
+} __packed;
+
+/**
+ * wifi_ps_txt - Get the power save setting as a text string
+ */
+static inline const char *wifi_ps_txt(enum wifi_ps ps)
+{
+	switch (ps) {
+	case WIFI_PS_DISABLED:
+		return "Power save disabled";
+	case WIFI_PS_ENABLED:
+		return "Power save enabled";
+	default:
+		return "Power save unknown";
+	}
+}
+
+enum wifi_ps_mode {
+	WIFI_PS_MODE_LEGACY = 0,
+	/* This has to be configured before connecting to the AP,
+	 * as support for ADDTS action frames is not available.
+	 */
+	WIFI_PS_MODE_WMM,
+} __packed;
+
+/**
+ * wifi_ps_mode_txt - Get the power save mode type as a text string
+ */
+static inline const char *wifi_ps_mode_txt(enum wifi_ps_mode ps_mode)
```

Comment:
it would be much simpler to have a const array of a char *.
Since all the enum here are from 0 to n iteratively. 

wifi_ps_mode2str[mode]  for instance, much less memory used. (no function code, just rom stuff)

also the wifi shell is the only one to use this, so move it there.

Reply:
As of now Wi-Fi shell is the only user to demonstrate the API use, but these functions are intended to be used samples and applications as well.

Reply:
Also, using const arrays handling defaults would be tough without adding a function, and IMHO, editing const arrays without explicit mapping is error prone (if we mismatch an enum).

Reply:
@tbursztyka Is the understanding here that the enums will be added incrementally and hence using a wifi_ps_mode2str suggested shouldn't be a problem or do we agree with @krish2718 comments and we can leave it as is ?

Reply:
If you need a function, make it a as a function. But don't make it inline (put it somewhere subsys/net/l2/wifi/wifi_helper.c for instance) and  get rid of the switch and have the const array instead.